cpp/src/reductions/reductions.cpp

```diff
@@ -38,128 +38,170 @@
 namespace cudf {
 namespace reduction {
 namespace detail {
-struct reduce_dispatch_functor {
-  column_view const col;
-  data_type output_dtype;
-  std::optional<std::reference_wrapper<scalar const>> init;
-  rmm::device_async_resource_ref mr;
-  rmm::cuda_stream_view stream;
-
-  reduce_dispatch_functor(column_view col,
-                          data_type output_dtype,
-                          std::optional<std::reference_wrapper<scalar const>> init,
-                          rmm::cuda_stream_view stream,
-                          rmm::device_async_resource_ref mr)
-    : col(std::move(col)), output_dtype(output_dtype), init(init), mr(mr), stream(stream)
-  {
+namespace {
+
+std::unique_ptr<scalar> reduce_aggregate_impl(
+  reduce_aggregation const& agg,
+  column_view col,
+  data_type output_dtype,
+  std::optional<std::reference_wrapper<scalar const>> init,
+  rmm::cuda_stream_view stream,
+  rmm::device_async_resource_ref mr)
```

Comment:
This code has not changed. The functor `operator()` was simply changed to a regular function call.
So the change just moved the code logic to the left.